Library/Homebrew/cmd/config.rb

```diff
@@ -119,9 +119,15 @@ def describe_system_ruby
     s
   end
 
-  def java_version
-    java = `java -version 2>&1`.lines.first.chomp
-    java =~ /java version "(.+?)"/ ? $1 : java
+  def describe_java
+    if which("java").nil?
+      "N/A"
+    elsif OS.mac? && `/usr/libexec/java_home --failfast` && !$?.success?
+      "N/A"
+    else
+      java = `java -version 2>&1`.lines.first.chomp
+      java =~ /java version "(.+?)"/ ? $1 : java
```

Comment:
If this is still slowish I'd consider just scraping it from the `java_home` output.